users/src/controllers/user.controller.ts

```diff
@@ -152,4 +164,65 @@ export const findRelation = async (req: Request, res: Response) => {
         status: relation
     }
     res.status(200).send(response);
+}
+
+/**
+ * Method for editing any field of the user with the specified id.
+ * Expects req.body to have a user object
+ * If all checks pass, will update User with the fields specified
+ * Then sends 200 alongside the newly updated user object.
+ * @param req 
+ * @param res 
+ */
+export const editUser = async (req: Request, res: Response) => {
+    if(!req.currentUser!.id) throw new BadRequestError("Please enter a user ID!");
+    console.log(req.currentUser!.id);
+    const user = req.user;
+    if(!user) throw new NotFoundError();
+    // if(isEmpty(req.body)) throw new BadRequestError("Empty request!");
+    if (req.body.year) {
+        const pattern = /^[1-4]$/
+        if (!pattern.test(req.body.year)) {
+            throw new BadRequestError('Year must be valid')
+        }
+    }
+    let imagePath = "";
+    try{
+        if (req.file) {
+            deleteFile(user.image);
+            imagePath = req.file.path;
+            console.log("imagePath: ", imagePath)
+            await user.update(
```

Comment:
move update logic outside if else clause

Reply:
Then how to handle file

Reply:
think about it, it's obvious